legend-engine-language-pure-dsl-service-execution/src/main/java/org/finos/legend/engine/language/pure/dsl/service/execution/AbstractServicePlanExecutor.java

```diff
@@ -122,19 +136,82 @@ protected AbstractServicePlanExecutor(String servicePath, String executionPlanRe
         this(servicePath, executionPlanResource, DEFAULT_ALLOW_JAVA_COMPILATION, storeExecutors);
     }
 
+    @Override
     public String getServicePath()
     {
         return this.servicePath;
     }
 
+    @Override
     public PlanExecutorInfo getPlanExecutorInfo()
     {
         return this.executor.getPlanExecutorInfo();
     }
 
+    @Override
+    public ResultType getResultType()
+    {
+        if (this.plan instanceof SingleExecutionPlan)
+        {
+            return ((SingleExecutionPlan) this.plan).rootExecutionNode.resultType;
+        }
+        else if(this.plan instanceof CompositeExecutionPlan)
+        {
+            List<ResultType> resultTypes = ((CompositeExecutionPlan) this.plan).executionPlans.values()
+                    .stream()
+                    .map(x -> x.rootExecutionNode.resultType)
+                    .collect(Collectors.toList());
+
+            Assert.assertFalse(resultTypes.isEmpty(), () -> "Composite plan without any actual plan?");
+            Assert.assertTrue(resultTypes.stream().allMatch(x -> x.getClass().equals(resultTypes.get(0).getClass())), () -> "Composite with different result types among plans?");
+            return resultTypes.get(0);
```

Comment:
@kevin-m-knight-gs : I added the validation here.  While every plan should yield the same result type, I'm hesitant to add the validation to plan  generation given the negative impact that can have if is not the case.   If we face issues with this API, we can then understand the reasoning and adapt better.  Thoughts?

Reply:
This validation is a little strange. We're only validating that all the result types are of the same class, not that they're all fully equal. I think full equality is the validation we really want to do. And I think we really should do it at plan generation time, not in this method. Since it's not the validation we actually want and it's not in the place we actually want it, let's just leave it off for now. You can revert this bit to what it was before.

However, we genuinely do need to understand whether the design is that the result type should be equal across all execution plans within a composite plan (even if the validations aren't fully in place to ensure it). If that is the case, then this API is fundamentally correct. But if it's possible to have different result types, then this API is fundamentally not correct. So we need to determine that before going forward.

Reply:
As discussed, after research, for relational the result type can be different on the tds relational column type.  Also, result type have editable fields that could impact the internal plan.  

For now, I will remove the getResultType() method until I can give more proper thoughts to how expose it.